external/storm-kafka/src/jvm/storm/kafka/PartitionManager.java

```diff
@@ -136,15 +136,15 @@ public EmitState next(SpoutOutputCollector collector) {
                 return EmitState.NO_EMITTED;
             }
             Iterable<List<Object>> tups = KafkaUtils.generateTuples(_spoutConfig, toEmit.msg);
-            if (tups != null) {
-		if(_spoutConfig.topicAsStreamId) {
-	            for (List<Object> tup : tups) {
-			collector.emit(_spoutConfig.topic, tup, new KafkaMessageId(_partition, toEmit.offset));
-		    }
-		} else {
-		    for (List<Object> tup : tups) {
-			collector.emit(tup, new KafkaMessageId(_partition, toEmit.offset));
-		    }
+            if ((tups != null) && tups.iterator.hasNext()) {
```

Comment:
This doesn't even compile.

>[ERROR] /home/travis/build/apache/storm/external/storm-kafka/src/jvm/storm/kafka/PartitionManager.java:[139,39] cannot find symbol
>
>  symbol:   variable iterator
>
>  location: variable tups of type java.lang.Iterable&lt;java.util.List&lt;java.lang.Object>>

I think it needs to be `tups.iterator().hasNext()`